sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -71,6 +71,23 @@ def test_kernel_pca_consistent_transform():
     assert_array_almost_equal(transformed1, transformed2)
 
 
+def test_kernel_pca_deterministic_output():
+    rng = np.random.RandomState(0)
+    X = rng.rand(10, 10)
+    eigen_solver = ('arpack', 'dense')
+    transformed_X = np.zeros((10 * len(eigen_solver), 2))
+
+    i = 0
+    for solver in eigen_solver:
+        for _ in range(10):
+            kpca = KernelPCA(n_components=2, eigen_solver=solver,
+                             random_state=i)
+            transformed_X[i, :] = kpca.fit_transform(X)[0]
+            i += 1
+
+    assert np.isclose(transformed_X, transformed_X[0, :]).all()
```

Comment:
Better to use  `numpy.testsing.assert_allclose`

Reply:
I find that undestand what is done in this test takes some effort here because we check equality for both different random seeds and solvers (and we store the results in a matrix that has non nstandard `(n_random_seeds*n_solvers, n_components)` shape. Maybe something like,
```
    for solver in eigen_solver: 
        transformed_X = np.zeros((10, 2))
        for i in range(10):
            kpca = KernelPCA(n_components=2, eigen_solver=solver,
                             random_state=i)
            transformed_X[i, :] = kpca.fit_transform(X)[0]

        assert_allclose(transformed_X, transformed_X[0, :])
```
could be easier?

Reply:
also is it needed to set the random state to i rather than just use `random_state=rng`?

Reply:
`random_state=rng` would use the same seed for all runs. The point here is to check that the arpack random state does not result in different solutions because of sign ambiguity

Reply:
I used `random_state=i` to show explicitly that the random state changes from one run to the next. But removing it would be valid as well as the random state would still change

Reply:
I may be wrong but i don't agree that `random_state=rng` would use the same seed, unless `rng` is an int. Here, `rng` is a `RandomState` object (initialized a few lines above) so it has internal state. For instance:

```python
~{main}$ ipython
Python 3.7.2 (default, Feb 11 2019, 11:12:39)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.2.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import numpy as np

In [2]: rng = np.random.RandomState(0)

In [3]: rng.randn(3)
Out[3]: array([1.76405235, 0.40015721, 0.97873798])

In [4]: rng.randn(3)
Out[4]: array([ 2.2408932 ,  1.86755799, -0.97727788])
```

Reply:
You're right, I can switch to that